sso-kit-client/react/src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,94 @@
+/*-
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full
+ * license.
+ */
+import type { ReactNode } from "react";
+import React from "react";
+import type { RouteObject } from "react-router-dom";
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import { useSsoContext } from "./useSsoContext.js";
+import type { AccessProps } from "../../core/src/AccessProps.js";
+
+interface ProtectedRouteProps {
+  redirectPath?: string;
+  route: ReactNode;
+}
+
+function ProtectedRoute({
+  redirectPath,
+  route,
+}: ProtectedRouteProps): JSX.Element | null {
+  const { authenticated, ssoContextInitialized } = useSsoContext();
+  const location = useLocation();
+
+  if (!ssoContextInitialized) {
+    return <div></div>;
+  }
+
+  if (!authenticated) {
+    if (redirectPath === undefined) {
+      redirectPath = "/ssologin";
+    }
+    return <Navigate to={redirectPath} state={{ from: location }} replace />;
+  }
+
+  return route ? (route as JSX.Element) : <Outlet />;
+}
+
+const collectRoutes = <T,>(routes: T[]): T[] => {
+  const allRoutes: T[] = [];
+  routes.forEach((route) => {
+    allRoutes.push(route);
+    if ((route as RouteObject).children !== undefined) {
+      allRoutes.push(...collectRoutes((route as RouteObject).children as T[]));
+    }
+  });
+  return allRoutes;
+};
+
+const protectRoute = <T,>(route: T, redirectPath?: string): void => {
+  if ((route as AccessProps).requireAuthentication) {
```

Comment:
Not big deal, but this is already checked by the caller.

Reply:
Nice catch, thank you. I will remove it in the next PR.